pkg/coffin/recover.go

```diff
@@ -10,12 +13,32 @@ func ResolveRecovery(unknownErr interface{}) error {
 		return nil
 
 	case error:
-		return rval
+		return withStack(rval)
 
 	case string:
-		return fmt.Errorf(rval)
+		return withStack(fmt.Errorf("%s", rval))
 
 	default:
-		return fmt.Errorf("unhandled error type %T", rval)
+		return withStack(fmt.Errorf("unhandled error type %T", rval))
 	}
 }
+
+func withStack(err error) error {
+	const depth = 32
+	var pcs [depth]uintptr
+
+	n := runtime.Callers(3, pcs[:])
+	st := runtime.CallersFrames(pcs[0:n])
+
+	stack := make([]string, 0, n)
+	done := false
+	for !done {
+		frame, more := st.Next()
+
+		stack = append(stack, fmt.Sprintf("%s:%d %s", frame.File, frame.Line, frame.Function))
+
+		done = !more
+	}
+
+	return errors.WithMessagef(err, "%s\nstack:\n\t%s", err.Error(), strings.Join(stack, "\n\t"))
```

Comment:
Can we do this with the built in errors package? 🤔

Reply:
Sure, I initially tried to avoid them because I had the feeling they were getting swallowed in some cases, but it was just the kernel shutting down as the last foreground module exited before the error could be propagated. Changed to `fmt`.